mixer/tools/adapterlinter/main.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"sort"
+	"strings"
+)
+
+var exitCode int
+
+var invalidImportPaths = map[string]string{
+	"log": "\"log\" import is not recommended; Adapters must instead use env.Logger for logging during execution. " +
+		"This logger understands about which adapter is running and routes the data to the place where the operator " +
+		"wants to see it.",
+	"github.com/golang/glog": "\"github.com/golang/glog\" import is not recommended; Adapters must instead use env.Logger for logging during execution. " +
+		"This logger understands about which adapter is running and routes the data to the place where the operator " +
+		"wants to see it.",
+}
+
+func main() {
```

Comment:
curious: do we want to handle `os.Signal` here? I feel like some of the gometalinter gripes I had in the past were related to having no real way to kill a started process.

Reply:
Hmm.It seems like gometalinter uses `Kill` https://github.com/alecthomas/gometalinter/blob/c89a599c3cf15a0034810515d6f922f64df2b6cd/execute.go#L171 to abort a process, which I hope kills a a process, irrespective of whether os.Signal stuff is handled. Also, it seems os.Signal will [not catch KILL commands](https://golang.org/pkg/os/signal/#hdr-Types_of_signals) . I might be wrong though.